Installation issue: blaspp with intel-oneapi-mkl

### Steps to reproduce the issue

```console
$ spack spec -I blaspp^intel-oneapi-mkl+cluster
Input spec
--------------------------------
 -   blaspp
 -       ^intel-oneapi-mkl+cluster

Concretized
--------------------------------
 -   blaspp@2022.07.00%gcc@11.2.0~cuda~ipo+openmp~rocm+shared build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen2
[+]      ^cmake@3.25.1%gcc@11.2.0~doc+ncurses+ownlibs~qt build_system=generic build_type=Release arch=linux-sles15-zen2
[+]          ^ncurses@6.3%gcc@11.2.0~symlinks+termlib abi=none build_system=autotools arch=linux-sles15-zen2
[+]              ^pkgconf@1.8.0%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]          ^openssl@1.1.1s%gcc@11.2.0~docs~shared build_system=generic certs=mozilla arch=linux-sles15-zen2
[+]              ^ca-certificates-mozilla@2022-10-11%gcc@11.2.0 build_system=generic arch=linux-sles15-zen2
[+]              ^perl@5.36.0%gcc@11.2.0+cpanm+shared+threads build_system=generic arch=linux-sles15-zen2
[+]                  ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-sles15-zen2
[+]                  ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared build_system=generic arch=linux-sles15-zen2
[+]                      ^diffutils@3.8%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                  ^gdbm@1.23%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                      ^readline@8.2%gcc@11.2.0 build_system=autotools patches=bbf97f1 arch=linux-sles15-zen2
[+]              ^zlib@1.2.13%gcc@11.2.0+optimize+pic+shared build_system=makefile arch=linux-sles15-zen2
[+]      ^intel-oneapi-mkl@2023.0.0%gcc@11.2.0+cluster+envmods~ilp64+shared build_system=generic arch=linux-sles15-zen2
[+]          ^intel-oneapi-tbb@2021.8.0%gcc@11.2.0+envmods build_system=generic arch=linux-sles15-zen2
[+]          ^mpich@4.0.2%gcc@11.2.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs+wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi patches=d4c0e99 pmi=pmi arch=linux-sles15-zen2
[+]              ^findutils@4.9.0%gcc@11.2.0 build_system=autotools patches=440b954 arch=linux-sles15-zen2
[+]              ^hwloc@2.8.0%gcc@11.2.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-sles15-zen2
[+]              ^libfabric@1.16.1%gcc@11.2.0~debug~kdreg build_system=autotools fabrics=sockets,tcp,udp arch=linux-sles15-zen2
[+]              ^libpciaccess@0.16%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                  ^libtool@2.4.7%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                  ^util-macros@1.19.3%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]              ^libxml2@2.10.3%gcc@11.2.0~python build_system=autotools arch=linux-sles15-zen2
[+]                  ^libiconv@1.16%gcc@11.2.0 build_system=autotools libs=shared,static arch=linux-sles15-zen2
[+]                  ^xz@5.2.7%gcc@11.2.0~pic build_system=autotools libs=shared,static arch=linux-sles15-zen2
[+]              ^yaksa@0.2%gcc@11.2.0~cuda~rocm build_system=autotools arch=linux-sles15-zen2
[+]                  ^autoconf@2.69%gcc@11.2.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-sles15-zen2
[+]                  ^automake@1.16.5%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                  ^m4@1.4.19%gcc@11.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-sles15-zen2
[+]                      ^libsigsegv@2.13%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                  ^python@3.10.8%gcc@11.2.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,f2fd060 arch=linux-sles15-zen2
[+]                      ^expat@2.5.0%gcc@11.2.0+libbsd build_system=autotools arch=linux-sles15-zen2
[+]                          ^libbsd@0.11.5%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                              ^libmd@1.0.4%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                      ^gettext@0.21.1%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz build_system=autotools arch=linux-sles15-zen2
[+]                          ^tar@1.34%gcc@11.2.0 build_system=autotools zip=pigz arch=linux-sles15-zen2
[+]                              ^pigz@2.7%gcc@11.2.0 build_system=makefile arch=linux-sles15-zen2
[+]                              ^zstd@1.5.2%gcc@11.2.0+programs build_system=makefile compression=none libs=shared,static arch=linux-sles15-zen2
[+]                      ^libffi@3.4.2%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
[+]                      ^libxcrypt@4.4.33%gcc@11.2.0~obsolete_api build_system=autotools arch=linux-sles15-zen2
[+]                      ^sqlite@3.40.0%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-sles15-zen2
[+]                      ^util-linux-uuid@2.38.1%gcc@11.2.0 build_system=autotools arch=linux-sles15-zen2
```

### Error message

<details><summary>Error message</summary><pre>
[+] /user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3
spack -e '/dev/shm/rmeli/dlaf-oneapi/packages/gcc-env' install   --only-concrete --only=package --no-add /mveydp43cusdwpic2h4atghzvhq7a3tu # blaspp@2022.07.00%gcc@11.3.0~cuda~ipo+openmp~rocm+shared build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen2
==> Installing blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu
==> No binary for blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu found: installing from source
==> Fetching https://mirror.spack.io/_source-cache/archive/56/566bd644f0364caffde6669e0f86514658eb06ca3d252a4fe67203921a875481.tar.gz
==> No patches needed for blaspp
==> blaspp: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/user-environment/linux-sles15-zen2/gcc-11.3.0/cmake-3.25.2-nz7t6wb2ocrk2otcwcwizibeei3aj3gc/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/user-environment/linux-sles15-zen2/gcc-11.3.0/blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/user-environment/linux-sles15-zen2/gcc-11.3.0/blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu/lib;/user-environment/linux-sles15-zen2/gcc-11.3.0/blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu/lib64;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/lib;/user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-binary-8.1.21.1-gcc-mkc6vuzultasziluk3k2dw2trfrpqn5h/lib;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-tbb-2021.8.0-fxkx6z7sr6berlm7aavkqlfju7sdl7cm/lib;/opt/cray/libfabric/1.15.0.0/lib64' '-DCMAKE_PREFIX_PATH:STRING=/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-tbb-2021.8.0-fxkx6z7sr6berlm7aavkqlfju7sdl7cm;/user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-binary-8.1.21.1-gcc-mkc6vuzultasziluk3k2dw2trfrpqn5h;/user-environment/linux-sles15-zen2/gcc-11.3.0/cmake-3.25.2-nz7t6wb2ocrk2otcwcwizibeei3aj3gc;/opt/cray/libfabric/1.15.0.0/' '-Dbuild_tests=False' '-Duse_openmp=True' '-DBUILD_SHARED_LIBS=True' '-Dgpu_backend=none' '-DBLAS_LIBRARIES=/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_scalapack_lp64.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_cdft_core.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_intel_lp64.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_sequential.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_core.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so;/usr/lib64/libpthread.so;/usr/lib64/libm.so;/usr/lib64/libdl.so' '/tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-mveydp43cusdwpic2h4atghzvhq7a3tu/spack-src'

1 error found in build log:
     16    -- Looking for BLAS libraries and options
     17    $BLAS_LIBRARIES
     18       libs:  /user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel
           64/libmkl_scalapack_lp64.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/
           2023.0.0/lib/intel64/libmkl_cdft_core.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtd
           ivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_intel_lp64.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.0.0-7rqug7vvgn
           urmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_sequential.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.
           0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_core.so;/user-environment/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-m
           kl-2023.0.0-7rqug7vvgnurmbhm5tw5kwtdivbc4ya3/mkl/2023.0.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so;/usr/lib64/libpthread.so;/usr/lib64/li
           bm.so;/usr/lib64/libdl.so
     19       -DBLAS_FORTRAN_ADD_                              no (library not found)
     20
     21       BLAS library not found.
  >> 22    CMake Error at CMakeLists.txt:374 (message):
     23      BLAS++ requires a BLAS library and none was found.  Ensure that it is
     24      accessible in environment variables $CPATH, $LIBRARY_PATH, and
     25      $LD_LIBRARY_PATH.
     26
     27
     28    -- Configuring incomplete, errors occurred!
</pre></details>


### Information on your system

```console
$ spack debug report
* **Spack:** 0.20.0.dev0 (25cb55ccd97d652d25e38a9a3aff1a456eee95b3)
* **Python:** 3.6.15
* **Platform:** linux-sles15-zen2
* **Concretizer:** clingo
```

### Additional information

<details><summary>spack-build-out.txt</summary><pre>
==> blaspp: Executing phase: 'cmake'
==> [2023-02-09-14:42:07.004401] '/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/cmake-3.25.1-6qjotyiex7o7krq2ujhcp7nafcsfkazf/bin/cmake' '-G' 'Unix Makefi
les' '-DCMAKE_INSTALL_PREFIX:STRING=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43' '-DCMAKE_BUILD_TYPE:
STRING=RelWithDebInfo' '-DBUILD_TESTING:BOOL=OFF' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_P
ATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/lib;/users/rme
li/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/lib64;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/int
el-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswa
isxuxlstnr/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/lib;/users/rmeli/spack/opt/spack/linux-sles15-ze
n2/gcc-11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0.16-6lplw7luj3uhhtygnpwwo5fygdl
grk7b/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2
/gcc-11.2.0/libiconv-1.16-kcfqnf5b2v5hfn7droowlcccikmtisqn/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2/lib
;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/
ncurses-6.3-hwwhiruax7hf63v6s4juzv26mpzde5w3/lib;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz/lib;/user
s/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw/lib' '-DCMAKE_PREFIX_PATH:STRING=/users/rmeli/spack/opt/spack/linux-sl
es15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcv
fzf7od75kiuev4c6;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw;/users/rmeli/spack/opt/spack/linux-sles15-zen2/g
cc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3;/us
ers/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.
2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0.16-6lplw7luj3uhhtygnpwwo5fygdlgrk7b;/users/rmeli/spack/o
pt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/cmake-3.25.
1-6qjotyiex7o7krq2ujhcp7nafcsfkazf;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex;/users/rmeli/spack/opt/spack
/linux-sles15-zen2/gcc-11.2.0/libiconv-1.16-kcfqnf5b2v5hfn7droowlcccikmtisqn;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/ncurses-6.3-hwwhiruax7hf63v6s4j
uzv26mpzde5w3' '-Dbuild_tests=False' '-Duse_openmp=True' '-DBUILD_SHARED_LIBS=True' '-Dgpu_backend=none' '-DBLAS_LIBRARIES=/users/rmeli/spack/opt/spack/linux-sles15-z
en2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_scalapack_lp64.so;/users/rmeli/spack/opt/spack/linux-sles15-
zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_cdft_core.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2
/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_intel_lp64.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gc
c-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_sequential.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-1
1.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_core.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/int
el-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so;/usr/lib64/libpthread.so;/usr/lib64/libm.so;/usr/lib64/
libdl.so' '/tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/spack-src'
-- The CXX compiler identification is GNU 11.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /users/rmeli/spack/lib/spack/env/gcc/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CMAKE_INSTALL_PREFIX = /users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43
-- Skipping CUDA search. gpu_backend = none
-- Not building CUDA support in BLAS++
-- Skipping HIP/ROCm search. gpu_backend = none
-- Not building HIP/ROCm support in BLAS++
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Looking for BLAS libraries and options
$BLAS_LIBRARIES
   libs:  /users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_scalapa
ck_lp64.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_cdft_c
ore.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_intel_lp64
.so;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_sequential.so
;/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_core.so;/users/r
meli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so;/u
sr/lib64/libpthread.so;/usr/lib64/libm.so;/usr/lib64/libdl.so
   -DBLAS_FORTRAN_ADD_                              no (library not found)

   BLAS library not found.
CMake Error at CMakeLists.txt:374 (message):
  BLAS++ requires a BLAS library and none was found.  Ensure that it is
  accessible in environment variables $CPATH, $LIBRARY_PATH, and
  $LD_LIBRARY_PATH.

-- Configuring incomplete, errors occurred!
See also "/tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/spack-build-mzw5cpd/CMakeFiles/CMakeOutput.log".
</pre></details>

<details><summary>spack-build-env.txt</summary><pre>
APPS=/apps/hohgant; export APPS
BASH_ENV=/usr/share/lmod/lmod/init/bash; export BASH_ENV
function BASH_FUNC__spack_shell_wrapper%%() {  for var in LD_LIBRARY_PATH DYLD_LIBRARY_PATH DYLD_FALLBACK_LIBRARY_PATH;
 do
 eval "if [ -n \"\${${var}-}\" ]; then export SPACK_$var=\${${var}}; fi";
 done;
 if [ -n "${ZSH_VERSION:-}" ]; then
 emulate -L sh;
 fi;
 _sp_flags="";
 while [ ! -z ${1+x} ] && [ "${1#-}" != "${1}" ]; do
 _sp_flags="$_sp_flags $1";
 shift;
 done;
 if [ -n "$_sp_flags" ] && [ "${_sp_flags#*h}" != "${_sp_flags}" ] || [ "${_sp_flags#*V}" != "${_sp_flags}" ]; then
 command spack $_sp_flags "$@";
 return;
 fi;
 _sp_subcommand="";
 if [ ! -z ${1+x} ]; then
 _sp_subcommand="$1";
 shift;
 fi;
 case $_sp_subcommand in
 "cd")
 _sp_arg="";
 if [ -n "$1" ]; then
 _sp_arg="$1";
 shift;
 fi;
 if [ "$_sp_arg" = "-h" ] || [ "$_sp_arg" = "--help" ]; then
 command spack cd -h;
 else
 LOC="$(spack location $_sp_arg "$@")";
 if [ -d "$LOC" ]; then
 cd "$LOC";
 else
 return 1;
 fi;
 fi;
 return
 ;;
 "env")
 _sp_arg="";
 if [ -n "$1" ]; then
 _sp_arg="$1";
 shift;
 fi;
 if [ "$_sp_arg" = "-h" ] || [ "$_sp_arg" = "--help" ]; then
 command spack env -h;
 else
 case $_sp_arg in
 activate)
 _a=" $@";
 if [ -z ${1+x} ] || [ "${_a#* --sh}" != "$_a" ] || [ "${_a#* --csh}" != "$_a" ] || [ "${_a#* -h}" != "$_a" ] || [ "${_a#* --help}" != "$_a" ]; then
 command spack env activate "$@";
 else
 stdout="$(command spack $_sp_flags env activate --sh "$@")" || return;
 eval "$stdout";
 fi
 ;;
 deactivate)
 _a=" $@";
 if [ "${_a#* --sh}" != "$_a" ] || [ "${_a#* --csh}" != "$_a" ]; then
 command spack env deactivate "$@";
 else
 if [ -n "$*" ]; then
 command spack env deactivate -h;
 else
 stdout="$(command spack $_sp_flags env deactivate --sh)" || return;
 eval "$stdout";
 fi;
 fi
 ;;
 *)
 command spack env $_sp_arg "$@"
 ;;
 esac;
 fi;
 return
 ;;
 "load" | "unload")
 _a=" $@";
 if [ "${_a#* --sh}" != "$_a" ] || [ "${_a#* --csh}" != "$_a" ] || [ "${_a#* -h}" != "$_a" ] || [ "${_a#* --list}" != "$_a" ] || [ "${_a#* --help}" != "$_a" ]; then
 command spack $_sp_flags $_sp_subcommand "$@";
 else
 stdout="$(command spack $_sp_flags $_sp_subcommand --sh "$@")" || return;
 eval "$stdout";
 fi
 ;;
 *)
 command spack $_sp_flags $_sp_subcommand "$@"
 ;;
 esac
}; export -f BASH_FUNC__spack_shell_wrapper%%
function BASH_FUNC_ml%%() {  eval $($LMOD_DIR/ml_cmd "$@")
}; export -f BASH_FUNC_ml%%
function BASH_FUNC_module%%() {  eval $($LMOD_CMD bash "$@") && eval $(${LMOD_SETTARG_CMD:-:} -s sh)
}; export -f BASH_FUNC_module%%
function BASH_FUNC_spack%%() {  : this is a shell function from: /users/rmeli/spack/share/spack/setup-env.sh;
 : the real spack script is here: /users/rmeli/spack/bin/spack;
 _spack_shell_wrapper "$@";
 return $?
}; export -f BASH_FUNC_spack%%
CC=/users/rmeli/spack/lib/spack/env/gcc/gcc; export CC
CLUSTER_NAME=hohgant; export CLUSTER_NAME
CMAKE_PREFIX_PATH=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn:/users/rmeli/spack/opt/spack/li
nux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7
fkzoaw:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-
11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6:/users/r
meli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0.1
6-6lplw7luj3uhhtygnpwwo5fygdlgrk7b:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr:/users/rmeli/s
pack/opt/spack/linux-sles15-zen2/gcc-11.2.0/cmake-3.25.1-6qjotyiex7o7krq2ujhcp7nafcsfkazf:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz
4fwgf5jle6nwvtds54m754kwex:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libiconv-1.16-kcfqnf5b2v5hfn7droowlcccikmtisqn:/users/rmeli/spack/opt/spack/linux
-sles15-zen2/gcc-11.2.0/ncurses-6.3-hwwhiruax7hf63v6s4juzv26mpzde5w3; export CMAKE_PREFIX_PATH
COLORTERM=1; export COLORTERM
CPU=x86_64; export CPU
CSHEDIT=emacs; export CSHEDIT
CXX=/users/rmeli/spack/lib/spack/env/gcc/g++; export CXX
DBUS_SESSION_BUS_ADDRESS=unix:path=/run/user/23000/bus; export DBUS_SESSION_BUS_ADDRESS
F77=/users/rmeli/spack/lib/spack/env/gcc/gfortran; export F77
FC=/users/rmeli/spack/lib/spack/env/gcc/gfortran; export FC
FROM_HEADER=''; export FROM_HEADER
GPG_TTY=/dev/pts/15; export GPG_TTY
G_BROKEN_FILENAMES=1; export G_BROKEN_FILENAMES
G_FILENAME_ENCODING=@locale,UTF-8,ISO-8859-15,CP1252; export G_FILENAME_ENCODING
HISTSIZE=1000; export HISTSIZE
HOME=/users/rmeli; export HOME
HOST=nid003192; export HOST
HOSTNAME=nid003192; export HOSTNAME
HOSTTYPE=x86_64; export HOSTTYPE
JAVA_BINDIR=/usr/lib64/jvm/java/bin; export JAVA_BINDIR
JAVA_HOME=/usr/lib64/jvm/java; export JAVA_HOME
JDK_HOME=/usr/lib64/jvm/java; export JDK_HOME
JRE_HOME=/usr/lib64/jvm/java; export JRE_HOME
LANG=en_US.UTF-8; export LANG
LC_ALL=C; export LC_ALL
LESS='-M -I -R'; export LESS
LESSCLOSE='lessclose.sh %s %s'; export LESSCLOSE
LESSKEY=/etc/lesskey.bin; export LESSKEY
LESSOPEN='lessopen.sh %s'; export LESSOPEN
LESS_ADVANCED_PREPROCESSOR=no; export LESS_ADVANCED_PREPROCESSOR
LIBGL_DEBUG=quiet; export LIBGL_DEBUG
LMOD_CMD=/usr/share/lmod/lmod/libexec/lmod; export LMOD_CMD
LMOD_DIR=/usr/share/lmod/lmod/libexec; export LMOD_DIR
LMOD_PKG=/usr/share/lmod/lmod; export LMOD_PKG
LMOD_SETTARG_FULL_SUPPORT=no; export LMOD_SETTARG_FULL_SUPPORT
LMOD_SYSTEM_NAME=hohgant; export LMOD_SYSTEM_NAME
LMOD_VERSION=8.3.1; export LMOD_VERSION
LMOD_sys=Linux; export LMOD_sys
LOGNAME=rmeli; export LOGNAME
LUSTRE_JOB_ID=hohgant_rmeli_nid003192; export LUSTRE_JOB_ID
MACHTYPE=x86_64-suse-linux; export MACHTYPE
MAIL=/var/spool/mail/rmeli; export MAIL
MANPATH=/usr/share/lmod/lmod/share/man:/usr/local/man:/usr/local/share/man:/usr/share/man:/usr/man; export MANPATH
MINICOM='-c on'; export MINICOM
MKLROOT=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0; export MKLROOT
MODULEPATH=/users/rmeli/spack/share/spack/modules/linux-sles15-zen2:/etc/cscs-modules; export MODULEPATH
MODULESHOME=/usr/share/lmod/lmod; export MODULESHOME
MORE=-sl; export MORE
MPICC=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/bin/mpicc; export MPICC
MPICH_CC=/users/rmeli/spack/lib/spack/env/gcc/gcc; export MPICH_CC
MPICH_CXX=/users/rmeli/spack/lib/spack/env/gcc/g++; export MPICH_CXX
MPICH_F77=/users/rmeli/spack/lib/spack/env/gcc/gfortran; export MPICH_F77
MPICH_F90=/users/rmeli/spack/lib/spack/env/gcc/gfortran; export MPICH_F90
MPICH_FC=/users/rmeli/spack/lib/spack/env/gcc/gfortran; export MPICH_FC
MPICXX=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/bin/mpic++; export MPICXX
MPIF77=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/bin/mpif77; export MPIF77
MPIF90=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/bin/mpif90; export MPIF90
OSTYPE=linux; export OSTYPE
PAGER=less; export PAGER
PATH=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/cmake-3.25.1-6qjotyiex7o7krq2ujhcp7nafcsfkazf/bin:/users/rmeli/spack/lib/spack/env/gcc:/users/rmeli/spa
ck/lib/spack/env/case-insensitive:/users/rmeli/spack/lib/spack/env:/users/rmeli/.spack/bootstrap/store/linux-centos7-x86_64/gcc-10.2.1/patchelf-0.16.1-p72zyan5wrzuabt
mzq7isa5mzyh6ahdp/bin:/users/rmeli/git/stackinator/bin/:/users/rmeli/spack/bin:/users/rmeli/.cargo/bin:/users/rmeli/node-v18.12.1-linux-x64/bin/:/users/rmeli/cmake-3.
25.1-linux-x86_64/bin/:/users/rmeli/.local/bin:/usr/local/bin:/usr/bin:/bin:/usr/lpp/mmfs/bin:/usr/lib/mit/bin; export PATH
PELOCAL_PRGENV=1; export PELOCAL_PRGENV
PKG_CONFIG_PATH=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/lib/pkgconfig:/users/rmeli/spack/opt/spack/linu
x-sles15-zen2/gcc-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopz
mi5lh2535gko6fkzner5hxz/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3/lib/pkgconfig:/users/rmel
i/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0
/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0.16-6lplw7luj3uhhtygnpwwo5fygdlgrk7b/
lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-
sles15-zen2/gcc-11.2.0/ncurses-6.3-hwwhiruax7hf63v6s4juzv26mpzde5w3/lib/pkgconfig:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-
rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/pkgconfig; export PKG_CONFIG_PATH
PROFILEREAD=true; export PROFILEREAD
PROJECT=/project/csstaff/rmeli; export PROJECT
#PWD=/users/rmeli; export PWD
QT_SYSTEM_DIR=/usr/share/desktop-data; export QT_SYSTEM_DIR
SCRATCH=/scratch/e1000/rmeli; export SCRATCH
SDK_HOME=/usr/lib64/jvm/java; export SDK_HOME
SHELL=/usr/local/bin/bash; export SHELL
SHLVL=1; export SHLVL
SPACK_CC=/opt/cray/pe/gcc/11.2.0/bin/gcc; export SPACK_CC
SPACK_CC_RPATH_ARG=-Wl,-rpath,; export SPACK_CC_RPATH_ARG
SPACK_COMPILER_FLAGS_KEEP=''; export SPACK_COMPILER_FLAGS_KEEP
SPACK_COMPILER_FLAGS_REPLACE='-Werror-|-Wno-error= -Werror|-Wno-error'; export SPACK_COMPILER_FLAGS_REPLACE
SPACK_COMPILER_IMPLICIT_RPATHS=/opt/cray/pe/gcc/11.2.0/snos/lib/gcc/x86_64-suse-linux/11.2.0:/opt/cray/pe/gcc/11.2.0/snos/lib64; export SPACK_COMPILER_IMPLICIT_RPATHS
SPACK_COMPILER_SPEC=gcc@11.2.0; export SPACK_COMPILER_SPEC
SPACK_CXX=/opt/cray/pe/gcc/11.2.0/bin/g++; export SPACK_CXX
SPACK_CXX_RPATH_ARG=-Wl,-rpath,; export SPACK_CXX_RPATH_ARG
SPACK_DEBUG_LOG_DIR=/users/rmeli; export SPACK_DEBUG_LOG_DIR
SPACK_DEBUG_LOG_ID=blaspp-mzw5cpd; export SPACK_DEBUG_LOG_ID
SPACK_DTAGS_TO_ADD=--disable-new-dtags; export SPACK_DTAGS_TO_ADD
SPACK_DTAGS_TO_STRIP=--enable-new-dtags; export SPACK_DTAGS_TO_STRIP
SPACK_ENV_PATH=/users/rmeli/spack/lib/spack/env:/users/rmeli/spack/lib/spack/env/case-insensitive:/users/rmeli/spack/lib/spack/env/gcc; export SPACK_ENV_PATH
SPACK_F77=/opt/cray/pe/gcc/11.2.0/bin/gfortran; export SPACK_F77
SPACK_F77_RPATH_ARG=-Wl,-rpath,; export SPACK_F77_RPATH_ARG
SPACK_FC=/opt/cray/pe/gcc/11.2.0/bin/gfortran; export SPACK_FC
SPACK_FC_RPATH_ARG=-Wl,-rpath,; export SPACK_FC_RPATH_ARG
SPACK_INCLUDE_DIRS=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/include:/users/rm
eli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/yaksa-
0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz/include:/user
s/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/li
bxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6/include/libxml2:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer
6/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/g
cc-11.2.0/libpciaccess-0.16-6lplw7luj3uhhtygnpwwo5fygdlgrk7b/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2w
tobb7sswaisxuxlstnr/tbb/2021.8.0/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex/include:/users/rmeli/s
pack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libiconv-1.16-kcfqnf5b2v5hfn7droowlcccikmtisqn/include:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/ncurses-6
.3-hwwhiruax7hf63v6s4juzv26mpzde5w3/include; export SPACK_INCLUDE_DIRS
SPACK_LINKER_ARG=-Wl,; export SPACK_LINKER_ARG
SPACK_LINK_DIRS=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64:/users/r
meli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv27f4rn/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11
.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw/lib:/users/
rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/hwl
oc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6/lib:/users/rme
li/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0
.16-6lplw7luj3uhhtygnpwwo5fygdlgrk7b/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr/tbb/2021
.8.0/lib/intel64/gcc4.8:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr/lib:/users/rmeli/spack/op
t/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libiconv-1.16-kcfqnf5b
2v5hfn7droowlcccikmtisqn/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/ncurses-6.3-hwwhiruax7hf63v6s4juzv26mpzde5w3/lib; export SPACK_LINK_DIRS
SPACK_PYTHON=/usr/bin/python3; export SPACK_PYTHON
SPACK_ROOT=/users/rmeli/spack; export SPACK_ROOT
SPACK_RPATH_DIRS=/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/lib:/users/rmeli/spack/opt/spack/linux-s
les15-zen2/gcc-11.2.0/blaspp-2022.07.00-mzw5cpdqwwexjmir5usefjp5ebzbgd43/lib64:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbk
xj4r73jutntf5u2kh6ouerv27f4rn/mkl/2023.0.0/lib/intel64:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-mkl-2023.0.0-rbkxj4r73jutntf5u2kh6ouerv2
7f4rn/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/mpich-4.0.2-4uecqgpja4dzpwcvfzf7od75kiuev4c6/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gc
c-11.2.0/yaksa-0.2-j5bmcqjjrzshgz7oq6ixhotts7fkzoaw/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libfabric-1.16.1-lhqngsopzmi5lh2535gko6fkzner5hxz/li
b:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/hwloc-2.8.0-q2zyfudvhqongiwqqnkajmi6kuibonb3/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0
/libxml2-2.10.3-ipkvok2brtefqmjzmcbze2gtnt6dfer6/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/xz-5.2.7-3x3rucnfhfdycxku7pnknj474hxgz2m2/lib:/users/rm
eli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/libpciaccess-0.16-6lplw7luj3uhhtygnpwwo5fygdlgrk7b/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/inte
l-oneapi-tbb-2021.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr/tbb/2021.8.0/lib/intel64/gcc4.8:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/intel-oneapi-tbb-2021
.8.0-eo7g37wszix2wtobb7sswaisxuxlstnr/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/zlib-1.2.13-73myhz4fwgf5jle6nwvtds54m754kwex/lib:/users/rmeli/spac
k/opt/spack/linux-sles15-zen2/gcc-11.2.0/libiconv-1.16-kcfqnf5b2v5hfn7droowlcccikmtisqn/lib:/users/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.2.0/ncurses-6.3-hwwh
iruax7hf63v6s4juzv26mpzde5w3/lib; export SPACK_RPATH_DIRS
SPACK_SHORT_SPEC='blaspp@2022.07.00%gcc@11.2.0~cuda~ipo+openmp~rocm+shared build_system=cmake build_type=RelWithDebInfo arch=linux-sles15-zen2/mzw5cpd'; export SPACK_
SHORT_SPEC
SPACK_SYSTEM_DIRS=/bin:/usr/bin:/usr/local/bin:/bin64:/usr/bin64:/usr/local/bin64:/include:/usr/include:/usr/local/include:/lib:/usr/lib:/usr/local/lib:/lib64:/usr/li
b64:/usr/local/lib64:/:/usr:/usr/local; export SPACK_SYSTEM_DIRS
SPACK_TARGET_ARGS='-march=znver2 -mtune=znver2'; export SPACK_TARGET_ARGS
SSH_CLIENT='148.187.134.13 57257 22'; export SSH_CLIENT
SSH_CONNECTION='148.187.134.13 57257 148.187.113.214 22'; export SSH_CONNECTION
SSH_TTY=/dev/pts/15; export SSH_TTY
TERM=xterm-256color; export TERM
USER=rmeli; export USER
WINDOWMANAGER=/usr/bin/gnome; export WINDOWMANAGER
XCURSOR_THEME=DMZ; export XCURSOR_THEME
XDG_CONFIG_DIRS=/etc/xdg; export XDG_CONFIG_DIRS
XDG_DATA_DIRS=/usr/share; export XDG_DATA_DIRS
XDG_RUNTIME_DIR=/run/user/23000; export XDG_RUNTIME_DIR
XDG_SESSION_CLASS=user; export XDG_SESSION_CLASS
XDG_SESSION_ID=315; export XDG_SESSION_ID
XDG_SESSION_TYPE=tty; export XDG_SESSION_TYPE
XKEYSYMDB=/usr/X11R6/lib/X11/XKeysymDB; export XKEYSYMDB
XNLSPATH=/usr/X11R6/lib/X11/nls; export XNLSPATH
</pre></details>




Linked to #22621? @rscohn2

`spack install blaspp ^intel-mkl` works as expected.

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

In my test locally with those same libraries outside Spack (below), it attempts to link with `libmkl_blacs_intelmpi_lp64.so` but is missing the MPI library. I find it odd that Spack's BLAS_LIBRARIES (i.e., `spec["blas"].libs.joined(";"),`) includes ScaLAPACK's BLACS — it seems ScaLAPACK should be a separate thing, not in BLAS. But if BLACS is in BLAS_LIBRARIES, then shouldn't MPI also be there? BLAS++ has no way to know that MPI should be added.

```
setenv BLAS "$MKLROOT/lib/intel64/libmkl_scalapack_lp64.so;$MKLROOT/lib/intel64/libmkl_cdft_core.so;$MKLROOT/lib/intel64/libmkl_intel_lp64.so;$MKLROOT/lib/intel64/libmkl_sequential.so;$MKLROOT/lib/intel64/libmkl_core.so;$MKLROOT/lib/intel64/libmkl_blacs_intelmpi_lp64.so"

cmake -DBLAS_LIBRARIES="${BLAS}" --log-level=trace ..
...
/nfs/apps/spacks/2023-01-01/opt/spack/linux-centos7-x86_64/gcc-9.5.0/intel-oneapi-mkl-2023.0.0-ahun4a3x3lfkmhp3hnztwufywmkyrqza/mkl/2023.0.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so: undefined reference to `MPI_Waitall'
...
```

Hello. I finally went back to this issue, and now
```bash
spack install blaspp ^intel-oneapi-mkl +cluster threads=none
```
seems to work as expected (with the latest `develop` Spack branch). I'm not yet sure what changed (I'll look into it), but things seems to work now.

However, FYI
```bash
spack install blaspp ^intel-oneapi-mkl +cluster threads=openmp
```
fails with similar issues, where `hello.cc` fails to compile:
```text
57 errors found in build log:
     133    gmake[1]: Entering directory '/tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-4zkfbwefz47dql3lel7aihsmad24dwib/spack-build-4zkfbwe/CMakeFiles/CMakeTmp'
     134    Building CXX object CMakeFiles/cmTC_e4fd4.dir/hello.cc.o
     135    /scratch/e1000/rmeli/spack/lib/spack/env/gcc/g++   -DBLAS_FORTRAN_ADD_   -fopenmp -o CMakeFiles/cmTC_e4fd4.dir/hello.cc.o -c /tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-4zkfbwefz47dql3lel7aihsmad24dwib/spack-s
            rc/config/hello.cc
     136    Linking CXX executable cmTC_e4fd4
     137    /user-environment/linux-sles15-zen2/gcc-11.3.0/cmake-3.26.3-evboqia5t3poocvdjrdsex4i4k3btein/bin/cmake -E cmake_link_script CMakeFiles/cmTC_e4fd4.dir/link.txt --verbose=1
     138    /scratch/e1000/rmeli/spack/lib/spack/env/gcc/g++ CMakeFiles/cmTC_e4fd4.dir/hello.cc.o -o cmTC_e4fd4  -Wl,-rpath,/scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2x
            jj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64:/user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjagneezuil/lib /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/in
            tel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_scalapack_lp64.so /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hd
            jeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_cdft_core.so /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_in
            tel_lp64.so /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so /scratch/e1000/rmeli/spack/opt/sp
            ack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_core.so /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-202
            3.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_blacs_intelmpi_lp64.so /user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjagneezuil/lib/libmpi.so /user-en
            vironment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjagneezuil/lib/libmpich.so /user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjagneezuil/lib/li
            bpmi.so /user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjagneezuil/lib/libopa.so /user-environment/linux-sles15-zen2/gcc-11.3.0/cray-mpich-8.1.23-gcc-4kfse26fsaibq23z23a5ahjag
            neezuil/lib/libmpl.so /usr/lib64/libpthread.so -lm /usr/lib64/libdl.so /user-environment/linux-sles15-zen2/gcc-11.3.0/gcc-11.3.0-a5sa43s4hwo76av4bylvwvrferv5mp3q/lib64/libgomp.so /usr/lib64/libpthread.so
  >> 139    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed8_orb_cpt'
  >> 140    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_next_4'
  >> 141    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            for_static_fini'
  >> 142    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_init_4u'
  >> 143    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed8_rd'
  >> 144    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            omp_task_alloc'
  >> 145    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_next_8'
  >> 146    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            barrier'
  >> 147    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_float4_wr'
  >> 148    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            omp_task'
  >> 149    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_init_4'
  >> 150    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_serialized_parallel'
  >> 151    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed4_add'
  >> 152    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed8_swp'
  >> 153    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed4_wr'
  >> 154    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_fini_8'
  >> 155    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            for_static_init_4u'
  >> 156    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed8_add'
  >> 157    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed8_wr'
  >> 158    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_next_4u'
  >> 159    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_float4_add'
  >> 160    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_float8_max'
  >> 161    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            reduce'
  >> 162    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            critical_with_hint'
  >> 163    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_critical'
  >> 164    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            for_static_init_8'
  >> 165    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_float8_add'
  >> 166    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            master'
  >> 167    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_reduce'
  >> 168    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            flush'
  >> 169    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_next_8u'
  >> 170    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            for_static_init_8u'
  >> 171    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            fork_call'
  >> 172    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            for_static_init_4'
  >> 173    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            ordered'
  >> 174    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            push_num_threads'
  >> 175    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_init_8u'
  >> 176    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_cmplx8_add'
  >> 177    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            global_thread_num'
  >> 178    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            critical'
  >> 179    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            omp_task_with_deps'
  >> 180    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            reduce_nowait'
  >> 181    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_float4_max'
  >> 182    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_master'
  >> 183    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            single'
  >> 184    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            dispatch_init_8'
  >> 185    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_cmplx4_add'
  >> 186    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            ok_to_fork'
  >> 187    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_single'
  >> 188    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_ordered'
  >> 189    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            end_reduce_nowait'
  >> 190    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            serialized_parallel'
  >> 191    /usr/bin/ld: /scratch/e1000/rmeli/spack/opt/spack/linux-sles15-zen2/gcc-11.3.0/intel-oneapi-mkl-2023.1.0-gpwr2rbpq74h2xjj5hdjeg2wcgb77grw/mkl/2023.1.0/lib/intel64/libmkl_intel_thread.so: undefined reference to `__kmpc_
            atomic_fixed4_rd'
  >> 192    collect2: error: ld returned 1 exit status
  >> 193    gmake[1]: *** [CMakeFiles/cmTC_e4fd4.dir/build.make:115: cmTC_e4fd4] Error 1
     194    gmake[1]: Leaving directory '/tmp/rmeli/spack-stage/spack-stage-blaspp-2022.07.00-4zkfbwefz47dql3lel7aihsmad24dwib/spack-build-4zkfbwe/CMakeFiles/CMakeTmp'
  >> 195    gmake: *** [Makefile:127: cmTC_e4fd4/fast] Error 2
     196
     197    >>>
     198       -DBLAS_FORTRAN_ADD_                              no (library not found)
     199
     200       BLAS library not found.
     201    --
```

--- a/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py
+++ b/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py
@@ -105,7 +105,7 @@ class IntelOneapiMkl(IntelOneApiLibraryPackage):
         multi=False,
     )
 
-    depends_on("intel-oneapi-tbb")
+    depends_on("tbb")
     # cluster libraries need mpi
     depends_on("mpi", when="+cluster")
 

--- a/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py
+++ b/var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py
@@ -105,7 +105,7 @@ class IntelOneapiMkl(IntelOneApiLibraryPackage):
         multi=False,
     )
 
-    depends_on("tbb")
+    depends_on("intel-oneapi-tbb")
     # cluster libraries need mpi
     depends_on("mpi", when="+cluster")
 
@@ -167,7 +167,10 @@ class IntelOneapiMkl(IntelOneApiLibraryPackage):
         if "threads=tbb" in self.spec:
             libs.append("libmkl_tbb_thread")
         elif "threads=openmp" in self.spec:
-            libs.append("libmkl_intel_thread")
+            if self.spec.satisfies("%oneapi") or self.spec.satisfies("%intel"):
+                libs.append("libmkl_intel_thread")
+            else:
+                libs.append("libmkl_gnu_thread")
         else:
             libs.append("libmkl_sequential")
 

